src/ionic-datepicker.provider.js

```diff
@@ -251,12 +245,16 @@ angular.module('ionic-datepicker.provider', [])
             text: $scope.mainObj.todayLabel,
             type: 'button_today',
             onTap: function (e) {
-              var today = new Date();
-              refreshDateList(new Date());
-              $scope.selctedDateEpoch = resetHMSM(today).getTime();
-              if (!$scope.mainObj.closeOnSelect) {
-                e.preventDefault();
-              }
+              var today = new Date($scope.today);
+              var today_obj = {
+                date: today.getDate(),
+                month: today.getMonth(),
+                year: today.getFullYear(),
+                day: today.getDay(),
+                epoch: today.getTime(),
+                disabled: false
```

Comment:
Because function $scope.dateSelected is expecting an object with "epoch" field.
I add other fields because they can be useful, maybe.